sdks/python/apache_beam/io/gcp/bigquery_file_loads.py

```diff
@@ -858,9 +859,13 @@ def _load_data(
             lambda x,
             deleting_tables: deleting_tables,
             pvalue.AsIter(temp_tables_pc))
-        | "RemoveTempTables/AddUselessValue" >> beam.Map(lambda x: (x, None))
+        # TableReference has no deterministic coder, but as this de-duplication
```

Comment:
From what I can tell, `parse_table_reference` is used to normalize and populate TableReference instances. They are essentially a "(project, dataset, table) tuple."
It should be possible to `beam.Map(lambda x: (x.tableId, x.datasetId, x.projectId)) | beam.Distinct() | beam.Map(lambda x: parse_table_reference(*x))`

cc: @pabloem

Reply:
Assuming exactly these three fields would not be as future proof.

Reply:
Ack. The official BigQuery client (which this code doesn't use) has a hashable wrapper for this.
https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.table.TableReference.html